hydra-node/test/Hydra/CryptoSpec.hs

```diff
@@ -33,6 +33,15 @@ specSigningKey =
     prop "can be generated" $ \(seedA, seedB) -> do
       seedA /= seedB
         ==> generateSigningKey seedA =/= generateSigningKey seedB
+    propCollisionResistant "arbitrary @(SigningKey HydraKey)" (arbitrary @(SigningKey HydraKey))
+
+propCollisionResistant :: (Show a, Eq a) => String -> Gen a -> Spec
+propCollisionResistant name gen =
```

Comment:
Avoid code duplication. we have the identical in `CardanoSpec`. We could move it into Util module